src/cell_gater/widgets/sample_widget.py

```diff
@@ -192,13 +190,13 @@ def _set_dropdown_marker_upperbound(self):
         region_props = self.model.regionprops_df
         if region_props is not None and len(region_props) > 0:
             self.upper_bound_marker_col.addItems(region_props.columns)
-            
-        #TODO set default to column before "X_centroid"
-        # This does not work
-        # if "X_centroid" in list(self.model.regionprops_df.columns):
-        #     self.upper_bound_marker_col.setCurrentIndex(
-        #         self.model.regionprops_df.columns.index("X_centroid")-1 )
-        
+            #set default to the last column before X_centroid
+            if "X_centroid" in region_props.columns:
+                default_index = self.upper_bound_marker_col.findText("X_centroid")
```

Comment:
I would not do findtext here but rather just take the index of the column in regionprops. Since it is a list being passed on and we do not do any ordering the order is the same.

Reply:
agree, replaced in recent commit